components/org.wso2.carbon.metrics.impl/src/test/java/org/wso2/carbon/metrics/impl/MetricApiTest.java

```diff
@@ -128,4 +127,34 @@ public void testGetMetricWithDifferentLevel() {
             Assert.assertThat(e, IsInstanceOf.instanceOf(IllegalArgumentException.class));
         }
     }
+
+    public void testDeleteMetrics() {
+        try {
+            MetricManager.meter("org.wso2.main.throughput", Level.INFO);
+            assertTrue("Error occurred while deleting the metric from the registry",
+                    metricService.removeMetric("org.wso2.main.throughput"));
+        } catch (Exception e) {
+            fail("Metric should be removed");
+        }
+    }
+
+    public void testDeleteMetricsWithAnnotatedName() {
+        try {
+            MetricManager.meter("org.wso2.main[+].sub.throughput", Level.INFO, Level.INFO);
+            assertTrue("Error occurred while deleting the metric from the registry",
+                    metricService.removeMetric("org.wso2.main.sub.throughput"));
```

Comment:
Annotated names create multiple metrics. Remove Metric API should accept the annotated name and remove relevant metrics.

Reply:
Created an issue #72